amethyst_xr/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+extern crate amethyst_assets;
+extern crate amethyst_core;
+extern crate amethyst_error;
+
+pub mod components;
+mod systems;
+
+use std::collections::BTreeMap;
+use std::sync::{Mutex, MutexGuard};
+
+use amethyst_core::bundle::SystemBundle;
+use amethyst_core::nalgebra::{UnitQuaternion, Vector3, Matrix4};
+use amethyst_core::specs::prelude::DispatcherBuilder;
+use amethyst_error::Error;
+
+use amethyst_assets::Loader;
+
+pub trait XRBackend: Send {
+    fn wait(&mut self);
+
+    fn get_new_trackers(&mut self) -> Option<Vec<(u32, TrackerCapabilities)>>;
+    fn get_removed_trackers(&mut self) -> Option<Vec<u32>>;
+
+    fn get_tracker_position(&mut self, index: u32) -> TrackerPositionData;
+
+    fn get_area(&mut self) -> Vec<[f32; 3]>;
+    fn get_hidden_area_mesh(&mut self) -> Vec<[f32; 3]>;
+
+    fn get_tracker_models(&mut self, index: u32) -> TrackerModelLoadStatus;
+
+    fn get_gl_target_info(&mut self, near: f32, far: f32) -> Vec<XRTargetInfo>;
```

Comment:
This is obviously very gl specific. While this is probably not a concern right now, we will have to handle multiple backends here.